.github/update_conda_lock.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+from os import environ
+from subprocess import run, DEVNULL, PIPE
+from sys import exit
+from re import match, search
+
+def _run(cmd_string, multiword_last_arg='', return_stdout=False, **kwargs):
+    cmd = cmd_string.split() + ([multiword_last_arg]
+            if multiword_last_arg else [])
+    if return_stdout:
+        return run(cmd, check=True, encoding='utf-8', stdout=PIPE,
+                **kwargs).stdout
+    else:
+        return run(cmd, check=True, **kwargs)
+
+def _get_env(env_name, required=True):
+    if env_name not in environ:
+        if required:
+            print('ERROR: Required environment variable not found: ' + env_name + '!')
+        return None
+    env_var = environ[env_name]
+    print('* ' + env_name + ': ' + env_var)
+    return env_var
+
+def _remove_conda_env(conda_env_name):
+    print('Removing `' + conda_env_name + '` Conda environment... ', end='')
+    _run('conda env remove -n ' + conda_env_name, stdout=DEVNULL,
+            stderr=DEVNULL)
+    print('done!')
+    print()
+
+# Returns True if lock file has been updated, False otherwise
+def try_updating_lock_file(path, new_lock):
+    print('Trying to update `' + path + '`...')
+    try:
+        with open(path, 'r') as f:
+            old_lock = f.read()
+            if old_lock == new_lock:
+                print(path + ' is up to date.')
+                print()
+                return False
+    except FileNotFoundError:
+        print(path + ' doesn\'t exist; it will be created.')
+    with open(path, 'w') as f:
+        f.write(new_lock)
+    print()
+    return True
+
+def main():
+    print('Environment variables used are:')
+    conda_env = _get_env('BOT_ENV_NAME')
+    environment_path = _get_env('BOT_ENV_YML')
+    conda_lock_path = _get_env('BOT_CONDA_LOCK')
+    pip_lock_path = _get_env('BOT_PIP_LOCK', required=False)
+    print()
+    if None in [conda_env, environment_path, conda_lock_path]:
+        exit(1)
+
+    print('Creating `' + conda_env + '` environment based on `'
+            + environment_path + '`...')
+    print()
+    try:
+        _run('conda env create -n ' + conda_env + ' -f ' + environment_path)
+    except:
+        print('ERROR: Creating `' + conda_env + '` environment failed!')
+        print('Please remove any environment with such name, if exists.')
+        print()
+        exit(1)
+
+    # Capture explicit list of Conda packages
+    conda_lock = _run('conda list --explicit -n ' + conda_env, return_stdout=True)
+    print('Conda packages captured.')
+    print()
+
+    # Capture frozen list of pip packages
+
+    # Variable will stay None in case there's no pip in environment.yml
+    pip_lock = None
+
+    # Check pip version
+    pip_version_match = search(r'pip\-([0-9]+)\.([0-9]+)[^\-]+-[^\-]+\.',
+            conda_lock)
+    if pip_version_match is not None:
+        if pip_lock_path is None:
```

Comment:
Can you explain (and add the revelant comments) around how `BOT_PIP_LOCK` is intended to be used?

Reply:
It was intended to be used after creating Conda environment. I found out I can merge `pip freeze` output into the locked `environment.yml` file so this file doesn't exist anymore.